contracts/utils/LocalTransferERC20.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+abstract contract LocalTransferERC20 {
```

Comment:
I feel this extraction is not worth it, unless we could also combine the use spender check to it. (in other words, the whole `_spendFromUserTo` body on RFQ)

But I have no good idea for the merge now 🥲 If other buddies have no ideas too, then I would suggest inlining `_localSpendFromUserTo` body in `_spendFromUserTo` on RFQ.

Reply:
It should probably at least incorporate the balance check.

For consumeGasERC20 check, since USDT/LINK/BNT are still very popular old ERC20 tokens but the list have not been updated ever, I think we can maybe update the list in constructor (hardcode the list or not) once and for all?

```solidity
constant consumeGasERC20 = [USDT_ADDR, LINK_ADDR, BNT_ADDR]

contract LocalTransferERC20 {
    constructor () {
        for addr in consumeGasERC20 {
            consumeGasERC20Tokens[addr] = true;
        }
    }
    function _localSpendFromUserTo(...) internal {
        ...
        uint256 gasStipend = consumeGasERC20Tokens[_tokenAddr] ? 80000 : gasleft();
        ...
    }
}
```

Reply:
what about token black list?

Reply:
The blacklist is never used.😂 I think we can remove it.